idflakies-core/src/main/java/edu/illinois/cs/dt/tools/detection/TestShuffler.java

```diff
@@ -234,4 +233,52 @@ public List<String> alphabeticalAndTuscanOrder(int count, boolean isTuscan) {
         }
         return fullTestOrder;
     }
+
+    public List<String> tuscanIntraClassOrder(int round) {
+        List<String> classes = new ArrayList<>(classToMethods.keySet());
+        HashMap<String, int[][]> classToPermutations = new HashMap<String, int[][]>();
+        Collections.sort(classes);
+        final List<String> fullTestOrder = new ArrayList<>();
+        int n = classes.size(); // n is number of classes
+        int[][] classOrdering = Tuscan.generateTuscanPermutations(n);
+        for (String className : classes) {
+            // This condition may be needed for the tests
+            // if (classToMethods.get(className).size() == 1) {
+            //     int[][] methodPermuation = { 
+            //         { 0, 0 }
+            //     };
+            //     classToPermutations.put(className, methodPermuation);
+            // } else {
+            int[][] methodPermuation = Tuscan.generateTuscanPermutations(classToMethods.get(className).size());
+            classToPermutations.put(className, methodPermuation);
+            // }
+        }
+        HashMap<String, List<String>> newClassToMethods = new HashMap<String, List<String>>();
+        List<String> permClasses = new ArrayList<String>();
+        int classRound = round;
+        while ((classOrdering.length - 1) < classRound) {
```

Comment:
Should add some comment to explain the reasoning/intention behind this loop concerning the class round, so we remember later what is the goal.